zerver/tests/test_bots.py

```diff
@@ -414,6 +416,39 @@ def test_bot_add_subscription(self) -> None:
         # Test runner automatically redirects all sent email to a dummy 'outbox'.
         self.assert_length(mail.outbox, 0)
 
+    def test_bot_add_user_group(self) -> None:
+        hamlet = self.example_user("hamlet")
+        self.login("hamlet")
+        params = {
+            "name": "support",
+            "members": orjson.dumps([hamlet.id]).decode(),
+            "description": "Support team",
+        }
+
+        self.client_post("/json/user_groups/create", info=params)
+        user_group = UserGroup.objects.get(name="support")
+
+        self.assertEqual(UserGroupMembership.objects.filter(user_group=user_group).count(), 1)
+
+        # Create a bot.
+        self.assert_num_bots_equal(0)
+        bot = self.create_bot()
+        self.assert_num_bots_equal(1)
+
+        params = {"add": orjson.dumps([bot["user_id"]]).decode()}
+
+        events_bot: List[Mapping[str, Any]] = []
+
+        with self.tornado_redirected_to_list(events_bot, expected_num_events=1):
+            result = self.client_post(f"/json/user_groups/{user_group.id}/members", info=params)
+            self.assert_json_success(result)
+
+        # No notification message is sent for adding a bot
+        msg_event = [e for e in events_bot if e["event"]["type"] == "message"]
+        self.assert_length(msg_event, 0)
+
+        self.assertEqual(UserGroupMembership.objects.filter(user_group=user_group).count(), 2)
```

Comment:
Would be good to add a test for checking case of removing bot from group, just for the sake of completion.

Reply:
Should I write a new test for removing bot from group or should I make changes to the test of adding bot to user group?